src/fs/rock/RockIoState.cc

```diff
@@ -210,7 +210,13 @@ Rock::IoState::tryWrite(char const *buf, size_t size, off_t coreOff)
             const auto sidNext = dir->reserveSlotForWriting(); // throws
             assert(sidNext >= 0);
             writeToDisk(sidNext);
-        } else if (Store::Root().transientReaders(*e)) {
+        } else if (false && Store::Root().transientReaders(*e)) {
+            // XXX: partial writes cause errors in SwapDir::droppedEarlierRequest(),
+            // sometimes getting unexpected '-1' slice id. We need to fix that bug
+            // before re-enabling this feature.
+            // XXX: the effective benefit of partial writes is limited by doPages()
+            // buffering (up to SM_PAGE_SIZE).
```

Comment:
I do not see buffering in doPages(). I see doPages() generating individual write requests, up to one page each, and stopping when there is no more bytes to write (or when there are errors/problems). Please double check that doPages() actually buffers (i.e. delays write requests even though there are more bytes to swap out and no errors/problems).

Reply:
For the record, SM_PAGE_SIZE is 4KB. The default db slot size is 16KB.

Reply:
doPages() has a do/while loop with the following lines in the end of the loop:

```
int64_t swapout_size = mem->endOffset() - mem->swapout.queue_offset;
if (anEntry->store_status == STORE_PENDING)
    if (swapout_size < SM_PAGE_SIZE)
        break;
```

So if there are swapout_size bytes to swapout but swapout_size<4KB, doPages() postpones writing on disk (caches in other words).

Reply:
So there is buffering of the last incomplete page, but only when that last incomplete page is preceded by some complete pages, right? For example,

* if doPages() is only given half a page, then it sends that half a page to disk. No buffering.
* if doPages() is given 1.5 pages, then it sends the first page to disk and pauses, buffering 0.5 pages. This buffering will end if another byte arrives (getting us back to the example bullet above!).

If the above description and examples are accurate, doPages() buffering is very strange and is probably not what its authors intended!

Reply:
> If the above description and examples are accurate

Accurate. That is probably a strange approach, but that is how it works.